### PR TITLE
Jonny reverse array solution

### PR DESCRIPTION
**Pull request by jonnyirving**

### CODE REVIEW

**Review by DavidSint — COMMENTED** [review]
Out of interest, which style would you choose to use on a client project?

I would probably go with your second one (with explanation comments on the lines before to explain the function), because your third one although cool, is not 'dev friendly' for the next person that comes along to read the code.

**Comment by GorillaGorilla** [comment]
The second one is a bit more readable I agree.

For a low level function that shouldn't really break I kind of think this doesn't matter too much, as long as its accompanied with tests which are readable and cover edge cases. That way they provide a plain english explanation of what it is supposed to do, and if it breaks what is going wrong.

One question I have is why you refer to arr[i] when you already have a reference to the current item? Force of habit?

File: reverseArray/JonnySolution.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+const countReverseReduce = (arr) => {
+	const count = arr.reduce((acc, item, i) => {
+  	const currentReversed = arr[i].split("").reverse().join("");
+	  return arr.includes(currentReversed) ? acc+1 : acc;
+  }, 0);
+  return count;
+}
+
+const countReverseReduceSmol = (arr) => (arr.reduce((acc, item, i) => (arr.includes(arr[i].split("").reverse().join("")) ? acc+1 : acc), 0));
```
Comment (anishamalde) [inline reverseArray/JonnySolution.js:20]: This is such a cool solution :)

**Review by anishamalde — APPROVED** [review]

**Comment by jonnyirving** [comment]
> Out of interest, which style would you choose to use on a client project?
> 
> I would probably go with your second one (with explanation comments on the lines before to explain the function), because your third one although cool, is not 'dev friendly' for the next person that comes along to read the code.



> Out of interest, which style would you choose to use on a client project?
> 
> I would probably go with your second one (with explanation comments on the lines before to explain the function), because your third one although cool, is not 'dev friendly' for the next person that comes along to read the code.

@DavidSint thanks for taking a look! Yeah, 100% the second one would be what I would go for generally on anything that would need to be maintained. The third one was just 'code golf' for fun.

**Comment by jonnyirving** [comment]
> The second one is a bit more readable I agree.
> 
> For a low level function that shouldn't really break I kind of think this doesn't matter too much, as long as its accompanied with tests which are readable and cover edge cases. That way they provide a plain english explanation of what it is supposed to do, and if it breaks what is going wrong.
> 
> One question I have is why you refer to arr[i] when you already have a reference to the current item? Force of habit?

Yeah this could have been the existing reference 👍
